e2e/LongHaul/module/IotHub.cs

```diff
@@ -259,6 +261,58 @@ public async Task SetPropertiesAsync(string keyName, object properties, Logger l
             }
         }
 
+        public async Task InvokeDirectMethodOnLeafClientThroughEdgeAsync(string deviceId, string moduleId, string methodName, Logger logger, CancellationToken ct)
+        {
+            Debug.Assert(_moduleClient != null);
+            Debug.Assert(deviceId != null);
+            Debug.Assert(moduleId != null);
+
+            var sw = new Stopwatch();
+            while (!ct.IsCancellationRequested)
+            {
+                try
+                {
+                    var payload = new CustomDirectMethodPayload
+                    {
+                        RandomId = Guid.NewGuid(),
+                        CurrentTimeUtc = DateTimeOffset.UtcNow,
+                        MethodCallsCount = ++_totalMethodCallsModuleToModuleSentCount,
+                    };
+
+                    var directMethodRequest = new EdgeModuleDirectMethodRequest(methodName, payload)
+                    {
+                        ResponseTimeout = TimeSpan.FromSeconds(30),
+                    };
+
+                    logger.Trace($"Invoking direct method for edge device: {deviceId}, edge module: {moduleId}", TraceSeverity.Information);
+                    logger.Metric(TotalDirectMethodCallsModuleToModuleSentCount, _totalMethodCallsModuleToModuleSentCount);
```

Comment:
This is also a premature metric, given we haven't successfully sent it yet, right?

Reply:
You're right. I moved both to be logged after the method invocation has completed.